stats.py

```diff
@@ -18,37 +19,32 @@ def find_extremes(weather_data):
             tuple: A tuple containing the highest temperature, highest temperature day,
                    lowest temperature, lowest temperature day, most humid day, and most humid day date.
         """
-        highest_temp = None
-        lowest_temp = None
-        most_humid_day = None
-        highest_temp_day = ""
-        lowest_temp_day = ""
-        most_humid_day_date = ""
+        extremes = {
+            "highest_temp": WeatherData(date="", max_temp=None, min_temp=None, mean_humidity=None),
```

Comment:
Why you are passing these defaults here? Keep these values as defaults in method definition